### DISCUSSION TITLE
書誌を更新するにあたって

### DISCUSSION THREAD

**Discussion by t-machii** [next-l/enju_leaf#1887] [Q&A]
お世話になっております。m(_ _)m

マニュアルのtsvで更新する、の項目
https://next-l.github.io/manual/1.4/enju_operation_4.html#section4-2-9
isbn などの識別子
> …注意！ISBNが複数ある場合、ISBNを用いて国立国会図書館のデータを参照して、 自動的に空白の項目が埋められることはありません。…

とあります。tsvにisbn と manifestation_id を記載し、更新したら自動的に空白が埋められる、と思っていたのですが実際にやってみると書誌にisbnが足され、書影が出るものの、他の空白の項目は埋まりません。
書誌に上書き、のような事はできないものでしょうか？

かわりに、isbnで書誌を作り、あらかじめ項目の埋まった書誌に所蔵をつけかえると元の書誌が残ります。
残ってしまった書誌は所蔵が無くてもtsvで一括削除はできない、という認識で正しいでしょうか？

また、後から作成した書誌に付け替えを行うと新着図書の上位に来てしまいます。
ソートを資料の 作成日 順ではなく 受入日 順を追加したり、所蔵のみにしたり…それをデフォルトに変更したり出来たら大変うれしいなと思いました。

まとまりのない文章で申し訳ありませんが、大量に書誌を更新したいと考えております。
何か、お知恵を拝借できれば幸いです。
よろしくお願い申し上げます。

**Comment by nabeta** [comment]
> tsvにisbn と manifestation_id を記載し、更新したら自動的に空白が埋められる

@t-machii 遅くなりましてすみません。これは、たとえば著者が入っていない書誌に対して、上記2項目を入力したTSVをアップロードして、著者を埋めたいということになりますでしょうか。

**Reply by nabeta** [reply]
> 残ってしまった書誌は所蔵が無くてもtsvで一括削除はできない、という認識で正しいでしょうか？

現状、アプリケーションには書誌の一括削除機能はありません。やるとすれば、以下の内容で`delete_manifestation.tsv`を作ってenju_leafディレクトリに保存し（`111`などは実際の削除対象のmanifestationのid）、

```
manifestation_id
111
222
333
```

コンソールで以下を実行することになります。ただし、所蔵や貸出情報が紐付いているものは削除できずエラーになるのでご注意ください。

```ruby
CSV.open('delete_manifestation.tsv', headers: true).each do |row|
  Manifestation.find(row['manifestation_id']).destroy
end
```

**Comment by t-machii** [comment]
@nabeta
いつもお返事ありがとうございます。
isbnと所蔵IDでのインポート、あまり件数があるとNDL側のアクセス過多？でエラーになってしまうので、
いったんタイトルのみで書誌・所蔵登録をしてしまったのです。
タイトルくらいしか入っておらず、ページ数や著者、件名、分類…空白が埋まればな、と思っていました。
tsvでの削除手順、ありがとうございます。これで大量の空書誌を消すことができそうです。
お礼申し上げます。
